### ISSUE TITLE
HTTP requests to VM services fail after extension update command

### ISSUE THREAD

**Issue by xenoscopic** [docker/extensions-sdk#224]
**Describe the bug**

Extensions which are updated using `docker extension update` seem to lose their ability to make requests to backend services.  Any backend request yields an error saying `connect ECONNREFUSED 127.0.0.1:80`.  The issue persists even after a switch away-from/back-to the extension pane, though it is resolved by restarting Docker Desktop entirely.

The issue only seems to occur with extension updates.  Extension removal followed by re-installation does not cause the issue.

**Add the steps to reproduce**

This behavior can be seen with the `vm-service` sample project (though it is also visible in other extensions that make requests to backend services).

1. Clone the `extensions-sdk` repo
2. Run `make install-extension` inside the `vm-service` sample directory
3. Open the newly installed extension pane. Volume information will be populated successfully from the backend service.
4. Run `make update-extension` inside the `vm-service` sample directory
5. Reload the extension pane by clicking out of it and then back into it.  No information will be populated.

The `vm-service` extension doesn't display errors by default, but they can be displayed by adjusting the `script.js` file in the project to the following:

```js
window.ddClient.extension.vm.service.get("/ls").then((volumes) => {
  document.body.innerHTML += `
    <ul>
      ${volumes.map((v) => `<li>${v}</li>`).join("")}
    </ul>
  `;
}).catch(function(e) {
  window.ddClient.desktopUI.toast.error(e.message);
});
```

With this change, an error message indicating `connect ECONNREFUSED 127.0.0.1:80` will be displayed.

**Describe the expected behavior**
The extension should update and continue to function without errors.  While pending requests may be preempted, future requests should not fail.

I would also not expect the client to attempt to connect to `127.0.0.1:80` given the connection forwarding mechanism that Docker Desktop is using to route requests to the backend socket.

**Output of `docker extension version`:**

```
$ docker extension version
Client Version: v0.2.13
Server API Version: 0.3.0
```

**Output of `docker version`:**

```
$ docker version
Client:
 Cloud integration: v1.0.29
 Version:           20.10.20
 API version:       1.41
 Go version:        go1.18.7
 Git commit:        9fdeb9c
 Built:             Tue Oct 18 18:20:35 2022
 OS/Arch:           darwin/amd64
 Context:           desktop-linux
 Experimental:      true

Server: Docker Desktop 4.13.1 (90346)
 Engine:
  Version:          20.10.20
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.7
  Git commit:       03df974
  Built:            Tue Oct 18 18:18:35 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.8
  GitCommit:        9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Additional context**

This issue seems to have been introduced in a recent version of Docker Desktop, I believe either 4.13.0 or 4.13.1.  It did not occur in earlier (but still recent) releases.

**Comment by felipecruz91** [comment]
Hi @xenoscopic, thank you for reporting the issue. We're aware of it and will be releasing a fix in the next version of Docker Desktop.

In the meantime, please use `docker extension rm` followed by `docker extension install`.

**Comment by xenoscopic** [comment]
Great, thanks for the quick response!

**Comment by xenoscopic** [comment]
Looks like this was fixed in Docker Desktop 4.14.0.  Thanks for the quick update!
